planner/core/stats.go

```diff
@@ -147,9 +149,142 @@ func (ds *DataSource) DeriveStats(childStats []*property.StatsInfo) (*property.S
 			break
 		}
 	}
+	// consider the IndexMergePath. Now, we just generate `IndexMergePath` in DNF case.
+	if len(ds.pushedDownConds) > 0 && len(ds.possibleAccessPaths) > 2 && ds.ctx.GetSessionVars().EnableIndexMerge {
```

Comment:
Since we consider table scan in index merge, the condition here should be `len(ds.possibleAccessPaths) > 1`. For example, if the filter is like `pk = 1 or a = 1`, and there are only one index `idx_a`, the current check would disable index merge.